experimental/PropertyDDS/packages/property-inspector-table/src/InspectorTable.tsx

```diff
@@ -262,715 +246,578 @@ class InspectorTable extends React.Component<WithStyles<typeof styles> & IInspec
-        this.setState(newState, () => {
-          if (forceUpdateRequired) {
-            this.forceUpdateBaseTable();
-          }
-          if (scrollingRequired) {
-            (this.table.current as any).scrollToRow(toExpand.rowIdx);
-          }
-        });
-      }
+  public componentDidUpdate(prevProps: ITableProps, prevState: IInspectorTableState) {
+    const { data, checkoutInProgress, followReferences } = this.props;
+    const { currentResult, expanded, tableRows, searchExpression, sortBy } = this.state;
+    let { foundMatches, childToParentMap } = this.state;
+    this.toTableRowOptions.followReferences = followReferences;
+    const newState = {} as Pick<IInspectorTableState, "currentResult" | "expanded" | "foundMatches" | "matchesMap" |
+      "searchAbortHandler" | "searchDone" | "searchInProgress" | "searchState" | "tableRows" | "childToParentMap">;
+
+    // Cancel all search activity and clear the search field when checking out a new repo.
+    if (checkoutInProgress && searchExpression && searchExpression.length > 0) {
+      this.handleOnClear();
     }
 
-    public render() {
-      const {
-        childGetter,
-        classes,
-        columns,
-        currentUrn,
-        data,
-        deleteRepo,
-        expired,
-        followReferences,
-        getRepoExpiry,
-        height,
-        isV1Urn,
-        nameGetter,
-        repositoryUrn,
-        rowIconRenderer,
-        searchBoxProps,
-        setRepoExpiry,
-        width,
-        activeRepositoryGuid,
-        ...restProps } = this.props;
-
-      const {
-        searchExpression,
-        expanded,
-        sortBy,
-        currentResult,
-        foundMatches,
-        searchInProgress,
-        editReferenceRowData,
-      } = this.state;
-
-      let rows = this.state.tableRows;
-
-      let expandedKeys: string[] = [];
-      let emptyDescription: string;
-      if (data !== undefined) {
-        const property = data.getProperty();
-        const workspace = property.getRoot().getWorkspace();
-        if (workspace) {
-          expandedKeys = Object.keys(expanded);
-          emptyDescription = InspectorMessages.EMPTY_WORKSPACE;
-        } else {
-          emptyDescription = InspectorMessages.NO_WORKSPACE;
-        }
-      } else {
-        rows = [];
-        emptyDescription = InspectorMessages.NO_DATA;
+    // We need to keep search data up to date with the current data.
+    // TODO: We cannot check if props.data changed, which is why we check props in general.
+    // This has the undesired side effect that we also restart search when the browser window is resized, for example.
+    if ((prevProps !== this.props || prevState.sortBy.order !== sortBy.order) && !checkoutInProgress) {
+      if (data) {
+        const updatedTableRows = this.props.toTableRows!({ data, id: "" }, this.props,
+          this.toTableRowOptions);
+        this.props.fillExpanded(expanded, updatedTableRows, this.props, this.toTableRowOptions);
```

Comment:
Here you don't check whether `fillExpanded` is provided. If it is correct, then you shouldn't check on line 289 as well